doc/application/index.rst

```diff
@@ -0,0 +1,12 @@
+.. _application_dev:
+
+Zephyr User Guide
+#################
+
+.. toctree::
+   :maxdepth: 1
+
+   ../porting/porting.rst
+   application.rst
+   ../samples/samples.rst
```

Comment:
Not sure hiding the samples documentation under "User Guide" is good, but without some doc usage stats we've got no data to help figure out what folks are finding important, and worthy of having easy access in the TOC/sidebar.